hazelcast-sql/src/main/java/com/hazelcast/sql/impl/schema/map/PartitionedMapTable.java

```diff
@@ -36,6 +36,7 @@ public class PartitionedMapTable extends AbstractMapTable {
     private final List<MapTableIndex> indexes;
     private final boolean hd;
     private final List<String> partitioningAttributes;
+    private final boolean supportsPartitionPruning;
```

Comment:
Not quite sure if it'd be better to just rename `partitioningAttributes` into `partitioningColumns` and include `__key` there. Having an explicit flag is clearer though.

Reply:
flag is ok. Some cases of `__key` do not support pruning, it might not be clear why sometimes `__key` is present and sometimes not.